### PR TITLE
Change the logic of the towers to only fire at the specfic color. Add…

### PR DESCRIPTION
**Pull request by Bluntsord**
Things changed:
1. Added sprites and levels for the towers. Thinking of maybe extending and having a resource to upgrade the towers in the future 
2. Added firing logic for towers. Towers now only fire at enemies of the same color

### CODE REVIEW

**Comment by Bluntsord** [comment]
no quantum logic has been implemented or added as of yet. Just the tower colors.

File: game/src/ui/gameplay/BuildTool/BuildUI.tscn [hunk 1]
```diff
@@ -30,6 +32,25 @@ margin_right = 12.1884
 margin_top = 13.5051
 margin_bottom = 14.4044
 
+[sub_resource type="GDScript" id=4]
```
Comment (bharathcs) [inline game/src/ui/gameplay/BuildTool/BuildUI.tscn]: delete this script

File: game/src/ui/gameplay/UI/UI.gd [hunk 2]
```diff
@@ -9,6 +9,7 @@ func can_drop_data(position, data):
 	return true
 
 func drop_data(position, data):
+	print(self.get_name())
```
Comment (bharathcs) [inline game/src/ui/gameplay/UI/UI.gd]: rm

File: game/src/ui/gameplay/BuildTool/BuildUI.gd [hunk 3]
```diff
@@ -128,3 +134,12 @@ func _pick_up_item(item: Item) -> void:
 	print(current_slot)		
 	print(build_UI_items_held)		
 
+
+func _on_TextureButton_pressed():
+	print("Called")
```
Comment (bharathcs) [inline game/src/ui/gameplay/BuildTool/BuildUI.gd]: rm

File: game/src/environment/map/tile-selector/TileSelector.gd [hunk 4]
```diff
@@ -43,6 +43,7 @@ func take_action():
 		return null
 	match (_action):
 		ACTION.BUILDING:
+			print("tower build from tile selector")
```
Comment (bharathcs) [inline game/src/environment/map/tile-selector/TileSelector.gd]: rm

File: game/src/environment/map/Map.gd [hunk 5]
```diff
@@ -72,6 +72,15 @@ func build_tower(x: int, y: int, tower_type: String) -> bool:
 		return false
 	# warning-ignore:unsafe_method_access
 	var tower : Tower = load(tower_type).instance()
+	var build_UI = self.get_node("UI/Control/BuildUI")
+	
+	#Check and change the universe of the towers
+
```
Comment (bharathcs) [inline game/src/environment/map/Map.gd:78]: remove unnecesary empty lines

**Review by bharathcs — APPROVED** [review]

**Pull request merged**